### PR TITLE
Autoupdate

### PR DESCRIPTION
**Pull request by romangraef**

### CODE REVIEW

**Comment by romangraef** [comment]
@IRONM00N can you merge this, infer is just picking up on all the other bugs in the same file that i havent caused (and probably arent critical either)

File: src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	public List<String> getCommandAliases() {
+		return Arrays.asList("neuupdates", "enoughupdates");
+	}
+
+	public void displayHelp(ICommandSender sender) {
+		sender.addChatMessage(new ChatComponentText(
+			"" +
+				"§e[NEU] §b/neuupdate help - View help.\n" +
+				"§e[NEU] §b/neuupdate check - Check for updates.\n" +
+				"§e[NEU] §b/neuupdate url <url> - Load an update from an direct download URL.\n" +
```
Comment (IRONM00N) [inline src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java:52]: this seems like a bad idea

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/updater/UpdateLoader.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+			return;
+		}
+		ArrayList<File> toDelete = new ArrayList<>();
+		File[] modFiles = mcDataDir.listFiles();
+		if (modFiles == null) {
+			updater.logProgress("§cCould not list minecraft mod folder (" + mcDataDir + ")");
+			state = State.FAILED;
+			return;
+		}
+		for (File sus : modFiles) {
```
Comment (IRONM00N) [inline src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/updater/UpdateLoader.java:110]: 🙄

**Review by IRONM00N — APPROVED** [review]

**Comment by IRONM00N** [comment]
> @IRONM00N can you merge this, infer is just picking up on all the other bugs in the same file that i havent caused (and probably arent critical either)

I would prefer if at least one other person approves first. Especially considering this is an auto updater.

File: src/main/java/io/github/moulberry/notenoughupdates/NEUOverlay.java [hunk 3]
```diff
@@ -1479,7 +1479,7 @@ public void updateSearch() {
 	 * Returns an index-able array containing the elements in searchedItems.
 	 * Whenever searchedItems is updated in updateSearch(), the array is recreated here.
 	 */
-	public List<JsonObject> getSearchedItems() {
+	public synchronized List<JsonObject> getSearchedItems() {
```
Comment (IRONM00N) [inline src/main/java/io/github/moulberry/notenoughupdates/NEUOverlay.java:1482]: Also, what do synchronized methods do?

File: src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java (hunk 1 above)
Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/commands/misc/UpdateCommand.java:52]: what?

**Comment by romangraef** [comment]
yeah i want other people to approve, just once there are approvals, can you merge this despite infer failing cause im not fixing all those alleged race conditions in the search bar

**Comment by IRONM00N** [comment]
> yeah i want other people to approve, just once there are approvals, can you merge this despite infer failing cause im not fixing all those alleged race conditions in the search bar

yea I can merge it then

**Review by NopoTheGamer — APPROVED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Review by DoKM — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Review by DeDiamondPro — APPROVED** [review]

**Review by DoKM — APPROVED** [review]

**Pull request merged**